src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,47 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        if (robot.getX() > toX) {
+            if (robot.getDirection() == Direction.UP) {
+                robot.turnLeft();
+            } else {
+                while (robot.getDirection() != Direction.LEFT) {
+                    robot.turnRight();
+                }
+            }
+        } else {
+            if (robot.getDirection() == Direction.UP) {
+                robot.turnRight();
+            } else {
+                while (robot.getDirection() != Direction.RIGHT) {
+                    robot.turnLeft();
+                }
+            }
+        }
```

Comment:
this is hard to understand what's going on here. try to use `switch/case` with robot's direction as a parameter

Reply:
done